File: problems/2588_count_the_number_of_beautiful_subarrays/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    long long beautifulSubarrays(vector<int>& nums) {
        unordered_map<int,int> cnt;
        cnt[0] = 1;
        int x = 0; long long ans = 0;
        for (int n : nums) { x ^= n; ans += cnt[x]; cnt[x]++; }
        return ans;
    }
};

File: problems/2588_count_the_number_of_beautiful_subarrays/solution.py
from typing import List


class Solution:
    def beautifulSubarrays(self, nums: List[int]) -> int:
        from collections import defaultdict
        cnt = defaultdict(int)
        cnt[0] = 1
        xor_sum = 0
        ans = 0
        for n in nums:
            xor_sum ^= n
            ans += cnt[xor_sum]
            cnt[xor_sum] += 1
        return ans
